engine/runtime/src/test/java/org/enso/interpreter/test/SignatureTest.java

```diff
@@ -727,6 +727,271 @@ public void andConversions() throws Exception {
     assertFalse("false & false", compute.execute(false, false).asBoolean());
   }
 
+  @Test
+  public void unresolvedReturnTypeSignature() throws Exception {
+    final URI uri = new URI("memory://neg.enso");
+    final Source src = Source.newBuilder("enso", """
+    neg a -> Xyz = 0 - a
+    """,uri.getAuthority())
+        .uri(uri)
+        .buildLiteral();
+
+    try {
+      var module = ctx.eval(src);
+      var neg = module.invokeMember(MethodNames.Module.EVAL_EXPRESSION, "neg");
+      fail("Expecting an exception from compilation, not: " + neg);
+    } catch (PolyglotException e) {
+      System.out.println(e);
+      assertTrue("It is a syntax error exception", e.isSyntaxError());
+    }
+  }
+
+  @Test
+  public void validReturnTypeSignature() throws Exception {
+    final URI uri = new URI("memory://rts.enso");
+    final Source src = Source.newBuilder("enso", """
+    from Standard.Base import Integer
+    add1 a b -> Integer = a+b
+    add2 (a : Integer) (b : Integer) -> Integer = a+b
+    """,uri.getAuthority())
+        .uri(uri)
+        .buildLiteral();
+
+    var module = ctx.eval(src);
+    var add1 = module.invokeMember(MethodNames.Module.EVAL_EXPRESSION, "add1");
+    assertEquals(3, add1.execute(1, 2).asInt());
+
+    var add2 = module.invokeMember(MethodNames.Module.EVAL_EXPRESSION, "add2");
+    assertEquals(3, add2.execute(1, 2).asInt());
```

Comment:
Possibly test with non-integer as well to demonstrate failure.

Reply:
This is checked by the test just below: `returnTypeCheckOptInError`.

Do you think I should merge these 2 tests into 1? I thought it may be clearer to keep the cases separate.